hmpps-prisoner-search-indexer/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/listeners/OffenderEventListenerIntTest.kt

```diff
@@ -33,6 +41,47 @@ class OffenderEventListenerIntTest : IntegrationTestBase() {
       val prisoner = prisonerRepository.get(prisonerNumber, listOf(SyncIndex.GREEN))
       assertThat(prisoner?.prisonerNumber).isEqualTo(prisonerNumber)
     }
+
+    verify(prisonerSpyBeanRepository, times(1)).save(any(), eq(SyncIndex.GREEN))
+    verify(prisonerSpyBeanRepository, never()).save(any(), eq(SyncIndex.BLUE))
+    verify(prisonerSpyBeanRepository, times(1)).save(any(), eq(SyncIndex.RED))
+    verify(prisonerSpyBeanRepository, never()).updateIncentive(eq(prisonerNumber), any(), eq(SyncIndex.RED), any())
+    verify(prisonerSpyBeanRepository, never()).updatePrisoner(eq(prisonerNumber), any(), eq(SyncIndex.GREEN), any())
+    verify(prisonerSpyBeanRepository, never()).updatePrisoner(eq(prisonerNumber), any(), eq(SyncIndex.RED), any())
+  }
```

Comment:
Calls save() for the red index if the prisoner does not exist, but updates if it does exist